core/server/types/objects.go

```diff
@@ -0,0 +1,57 @@
+package types
+
+import (
+	"bytes"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	kustomizev1 "github.com/fluxcd/kustomize-controller/api/v1beta2"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta2"
+	pb "github.com/weaveworks/weave-gitops/pkg/api/core"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// This method whitelists legal object kinds to request
+func GetGVK(kind string) *schema.GroupVersionKind {
```

Comment:
IMO, it would be better to handle the error at a function, which is possible to return `nil`
so that semantic of the error belongs to the function.

`func GetGVK(kind string) (*schema.GroupVersionKind, error) {`